catalyst/callbacks/batch_transform.py

```diff
@@ -1,171 +1,232 @@
+                    dl.CheckpointCallback(
+                        logdir="./logs", loader_key="valid", metric_key="map01", minimize=False
+                    ),
+                ]
+            )
 
-                if self.is_train_loader:
-                    loss.backward()
-                    self.optimizer.step()
-                    self.optimizer.zero_grad()
-
-        model = torch.nn.Linear(28 * 28, 10)
-        optimizer = torch.optim.Adam(model.parameters(), lr=0.02)
-
-        loaders = {
-            "train": DataLoader(
-                MNIST(os.getcwd(), train=True, transform=ToTensor()),
-                batch_size=32,
-            ),
-            "valid": DataLoader(
-                MNIST(os.getcwd(), train=False, transform=ToTensor()),
-                batch_size=32,
-            ),
-        }
-        transforms = [
-            augmentation.RandomAffine(degrees=(-15, 20), scale=(0.75, 1.25)),
-        ]
-
-        runner = CustomRunner()
-
-        # model training
-        runner.train(
-            model=model,
-            optimizer=optimizer,
-            loaders=loaders,
-            logdir="./logs",
-            num_epochs=5,
-            verbose=True,
-            callbacks=[BatchTransformCallback(transforms, input_key=0)],
-        )
-
-    To apply augmentations only during specific loader e.g. only during
-    training :class:`catalyst.core.callbacks.control_flow.ControlFlowCallback`
-    callback can be used. For config API it can look like this:
-
-    .. code-block:: yaml
-
-        callbacks_params:
-          ...
-          train_transforms:
-            _wrapper:
-              name: ControlFlowCallback
-              loaders: train
-            name: BatchTransformCallback
-            transform:
-              - transform: kornia.RandomAffine
-                degrees: [-15, 20]
-                scale: [0.75, 1.25]
-                return_transform: true
-              - transform: kornia.ColorJitter
-                brightness: 0.1
-                contrast: 0.1
-                saturation: 0.1
-                return_transform: false
-            input_key: image
-          ...
-
-    .. _`Kornia: an Open Source Differentiable Computer Vision Library
-        for PyTorch`: https://arxiv.org/pdf/1910.02190.pdf
     """
 
     def __init__(
         self,
-        transform: Sequence[Union[dict, nn.Module]],
-        input_key: Union[str, int] = "image",
-        output_key: Optional[Union[str, int]] = None,
-    ) -> None:
-        """Init."""
-        super().__init__(order=CallbackOrder.Internal, node=CallbackNode.all)
+        lambda_fn: Callable,
+        scope: str,
+        input_key: Union[List[str], str, int] = None,
+        output_key: Union[List[str], str, int] = None,
+    ):
+        """
+        Preprocess your batch with specified function.
 
+        Args:
+            lambda_fn (Callable): Function to apply.
+            scope (str): ``"on_batch_end"`` or ``"on_batch_start"``
```

Comment:
could you please add a small docstring, that `on_batch_start` is about batch preprocessing for the model, while `on_batch_end` - about the post processing

Reply:
it would be also great if we could return the kornia example for the preprocessing case :)